WebHostLib/downloads.py

```diff
@@ -72,7 +72,14 @@ def download_slot_file(room_id, player_id: int):
                     if name.endswith("info.json"):
                         fname = name.rsplit("/", 1)[0] + ".zip"
         elif slot_data.game == "Ocarina of Time":
-            fname = f"AP_{app.jinja_env.filters['suuid'](room_id)}_P{slot_data.player_id}_{slot_data.player_name}.apz5"
+            stream = io.BytesIO(slot_data.data)
+            if zipfile.is_zipfile(stream):
+                with zipfile.ZipFile(stream) as zf:
```

Comment:
did you test this? we currently file.seek(0) in upload.py line 128 as is_zipfile can leave the data stream partially read. I'd be surpised if that doesn't happen here.

Reply:
I did test it and it seemed to work fine (the game patched and played properly)

Reply:
interesting. well, let's see if it breaks for anyone.

Reply:
maybe I should just seek to 0 or make a new stream to be safe though